codec/reflectcodec/type_codec.go

```diff
@@ -523,6 +523,20 @@ func (c *genericCodec) Unmarshal(bytes []byte, dest interface{}) error {
 	return nil
 }
 
+// UnmarshalFrom unmarshals [p.Bytes] into [dest], where [dest] must be a pointer or
+// interface
+func (c *genericCodec) UnmarshalFrom(p *wrappers.Packer, dest interface{}) error {
```

Comment:
This looks to be identical to `Unmarshal` with the exception of not enforcing the offset == len(bytes) at the end of unmarshalling.

One small improvement would be to re-use this code in `Unmarshal` to avoid duplicating the same functionality.

I think that would just be `UnmarshalFrom(wrappers.Packer{bytes}, dest)` and then enforce the offset == length check that exists in `Unmarshal` but not `UnmarshalFrom`